detekt/src/main/kotlin/potentialbugs/ImmutableTypesWithExpectedInteriorMutabilityInConfig.kt

```diff
@@ -0,0 +1,86 @@
+package at.hannibal2.skyhanni.detektrules.potentialbugs
+
+import at.hannibal2.skyhanni.detektrules.SkyHanniRule
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Severity
+import io.gitlab.arturbosch.detekt.api.internal.RequiresTypeResolution
+import io.gitlab.arturbosch.detekt.rules.fqNameOrNull
+import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.psi.KtAnnotated
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtTypeReference
+import org.jetbrains.kotlin.resolve.typeBinding.createTypeBinding
+import org.jetbrains.kotlin.resolve.typeBinding.createTypeBindingForReturnType
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.types.typeUtil.immediateSupertypes
+
+@RequiresTypeResolution
+class ImmutableTypesWithExpectedInteriorMutabilityInConfig(config: Config) : SkyHanniRule(config) {
+	override val issue: Issue
+		get() = Issue(
+			"ImmutableTypesWithExpectedInteriorMutabilityInConfig",
+			Severity.Defect,
+			"Disallow using types which may disallow MoulConfig from mutating them, due to the ",
+			Debt.FIVE_MINS,
+		)
+
+	companion object {
+		val configPackage = FqName("at.hannibal2.skyhanni.config")
```

Comment:
We want to use immutable types in the storage part of the config.
Do you need to change this to `at.hannibal2.skyhanni.config.features`?

Reply:
The config package here is just a short circuit. Each individual property will also be checked for an `@ConfigOption` property:

https://github.com/nea89o/SkyHanni/blob/e270bdadc933dee713e5acd32046a49561b763df/detekt/src/main/kotlin/potentialbugs/ImmutableTypesWithExpectedInteriorMutabilityInConfig.kt#L71-L73